### PR TITLE
feat: implement --html and --xml output format options (#350)

### LINKED ISSUE #350: Implement --html and --xml output format options
## Overview

Implement HTML and XML output format options, compatible with Google Cloud Spanner CLI.

## Specification

- `--html`: Show output in HTML format
- `--xml`: Show output in XML format
- Should work for query results output

## Implementation notes

- spanner-mycli already supports table format (--table)
- Need to add HTML and XML formatters
- Consider integration with existing output formatting system
- Should handle special characters properly (escaping)

## Priority

Low priority - specialized output formats with limited use cases

## References

- Parent issue: #342
- [Google Cloud Spanner CLI documentation](https://cloud.google.com/spanner/docs/spanner-cli)

### PR DESCRIPTION
**Pull request by apstndb**
## Summary

- Implements `--html` and `--xml` output format options for query results (Fixes #350)
- Adds proper HTML/XML escaping for security (unlike the reference implementation)
- Maintains compatibility with Google Cloud Spanner CLI output structure

### Key Decisions

1. **Security-First Approach**: Unlike `spannercli sql`, which does NOT escape special characters in HTML/XML output, this implementation uses proper escaping:
   - HTML: `html.EscapeString()` for all values
   - XML: `encoding/xml` package handles escaping automatically
   
2. **Standard Libraries**: Used Go's built-in `html` and `encoding/xml` packages rather than custom string escaping for maintainability and correctness.

3. **Mutual Exclusivity**: Enforced that `--table`, `--html`, and `--xml` flags are mutually exclusive at the CLI level.

4. **Runtime Format Changes**: Extended `CLI_FORMAT` system variable to support `HTML` and `XML` values for runtime format switching.

5. **Code Simplicity**: After initial over-engineering with a complex `tableFormatter` abstraction, reverted to simple, direct implementations following the Go idiom "a little copying is better than a little dependency".

### Testing Notes

- Verified behavior with real Spanner instance
- Confirmed `--skip-column-names` works correctly with both HTML and XML formats
- Added comprehensive tests including special character escaping and edge cases
- All existing tests continue to pass

## Test plan

- [x] Run `make check` - all tests pass
- [x] Manual testing with `spanner-mycli --html` and `--xml` flags
- [x] Verify escaping with queries containing special characters
- [x] Test `SET CLI_FORMAT = 'HTML';` and `SET CLI_FORMAT = 'XML';`
- [x] Verify `--skip-column-names` compatibility

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: cli_output.go [hunk 1]
```diff
@@ -501,3 +525,104 @@ var (
...
+	for _, row := range rows {
+		fmt.Fprint(out, "<TR>")
+		for _, col := range row {
+			fmt.Fprintf(out, "<TD>%s</TD>", html.EscapeString(col))
+		}
+		fmt.Fprint(out, "</TR>")
+	}
+	
+	fmt.Fprintln(out, "</TABLE>")
+}
```
[the remark this answers is not in the document]
Comment (apstndb) [reply cli_output.go:558]: I investigated the error handling pattern throughout the codebase and found that the current implementation is consistent with the established patterns:

1. **No output functions return errors**: All print functions in `cli_output.go` (`printTableData`, `printResult`, etc.) have the same signature - they don't return errors.

2. **Errors are logged but execution continues**: When operations can fail (like `table.Render()` at line 112 or `template.Execute()` at line 273), errors are logged with `slog.Error()` but the function continues.

3. **Simple I/O operations don't check errors**: Throughout the codebase, `fmt.Fprintf` and `fmt.Fprintln` calls don't check for errors (see lines 127, 138, 141, 147, 150, etc.).

The `printHTMLTable` function follows this exact pattern - it doesn't return an error and doesn't check I/O errors, maintaining consistency with the rest of the codebase.

Changing this function to return errors would make it inconsistent with all other output functions and would require changing the entire output layer's error handling approach, which is beyond the scope of this PR.
EOF < /dev/null

**Comment by apstndb** [comment]
## Implementation Insights

### Design Journey: Simplicity Over Abstraction
During implementation, I initially created a complex `tableFormatter` abstraction to unify HTML and XML output logic. However, after reflection, I reverted to simple, direct implementations following the Go idiom "a little copying is better than a little dependency." This decision was driven by:
- The abstraction added complexity without significant benefit
- The direct implementations are clearer and easier to understand
- Each format has unique requirements that made abstraction awkward

### XML Character Escaping Discovery
Testing revealed that Go's `encoding/xml` package DOES escape quotes in character data (both single and double quotes become `&#39;` and `&#34;`). This was initially questioned in review but confirmed through testing. This behavior ensures proper XML safety.

### Security Enhancement Decision
Compatibility testing with the reference implementation (`spannercli sql`) revealed that it does NOT escape special characters in HTML/XML output, creating potential security vulnerabilities. We made the conscious decision to prioritize security over exact compatibility by implementing proper escaping.

### Error Handling Philosophy
The output functions follow the established "log and continue" pattern in the codebase rather than returning errors. This is documented in `.gemini/styleguide.md` to prevent future confusion. The rationale is that partial output is often more useful than no output in an interactive CLI tool.

**Comment by apstndb** [comment]
## Error Handling Update

Based on review feedback, I've updated the HTML and XML output functions to properly return errors:

- `printHTMLTable` and `printXMLResultSet` now return `error`
- All I/O operations check and propagate errors
- Errors are logged at the call site but execution continues (maintaining consistency with current behavior)
- Added TODO(#388) comment to migrate other output formats in a future PR
- Created issue #388 to track error handling improvements for other output formats

This sets a better pattern for future improvements while keeping the change scope limited to this PR.

The styleguide documentation about error handling patterns has been removed since we're moving toward proper error handling rather than documenting the current state as a best practice.

File: cli_output.go [hunk 2]
```diff
@@ -501,3 +532,131 @@ var (
...
+// printXMLResultSet outputs query results in XML format.
+// The format is compatible with Google Cloud Spanner CLI:
+//   - Uses XML declaration with single quotes: <?xml version='1.0'?>
+//   - Includes xmlns:xsi namespace for compatibility
+//   - All values are automatically XML-escaped by encoding/xml package
+// Note: This implementation builds the entire result set in memory before
+// encoding. For very large result sets, consider using TAB format for
+// streaming output. This design prioritizes simplicity and consistency
+// with the TABLE format over memory efficiency.
+func printXMLResultSet(out io.Writer, columnNames []string, rows []Row, skipColumnNames bool) error {
```
[the remark this answers is not in the document]
Comment (apstndb) [reply cli_output.go:620]: This concern has already been addressed in our previous discussion:

1. **Design consistency**: The XML implementation intentionally follows the same pattern as the TABLE format, which also buffers in memory (see lines 72-116 in cli_output.go).

2. **Explicitly documented**: The function comment at line 609-612 already states:
   > "Note: This implementation builds the entire result set in memory before encoding. For very large result sets, consider using TAB format for streaming output. This design prioritizes simplicity and consistency with the TABLE format over memory efficiency."

3. **Interactive CLI context**: spanner-mycli is an interactive CLI tool, not a bulk export utility. Users needing to export very large datasets would use dedicated tools or the TAB format (which already streams).

4. **Streaming HTML vs buffering XML**: While HTML can be streamed easily, XML with proper structure (including the header element) is more complex to stream correctly. The current implementation ensures well-formed XML output.

The review also mentions "bulk load feature" and "signed images to a registry" which are unrelated to this PR, suggesting there may be some confusion in the review.

The current implementation is intentional, documented, and consistent with existing patterns in the codebase.
EOF < /dev/null
